src/particle_filter/include/particle_filter/distanceTransform.h

```diff
@@ -0,0 +1,177 @@
+#ifndef DISTANCE_TRANSFORM_H
+#define DISTANCE_TRANSFORM_H
+#include <limits>
+
+class distanceTransform
+{
+ public:
+  double voxel_size; //voxel size
+  double world_range[3][2]; //work space range
+  double ***dist_transform;
+  double ***obstacle_map;
+  distanceTransform(double range, double voxel_size);
+  void build(double cube_para[3], double World_Range[3][2]);
+ protected:
+  int voxel_nums[3]; // voxel number along each dimension
+  void distanceTransform_1D(double ***dist_transform_1D, int range[3], double ***cost_fun, int dir, int idx1, int idx2);
+};
+/*distanceTransform::distanceTransform(double World_Range[3][2], double Grid_Res) :voxel_size(Grid_Res) {
+  memcpy(world_range, World_Range, 6 * sizeof(double));
+  }*/
+
+distanceTransform::distanceTransform(double range, double grid_Res): voxel_size(grid_Res) {
+  for (int i = 0; i < 3; i++) {
+    voxel_nums[i] = ceil(range / voxel_size);
+  }
+  dist_transform = new double **[voxel_nums[0]];
```

Comment:
You are using a lot of "new" commands with no corresponding "delete".
I suggest std::vector instead. If not, include the delete command in the destructor.